src/dataregistry/registrar/dataset.py

```diff
@@ -362,48 +365,113 @@ def _handle_data(self, relative_path, old_location, owner, owner_type, verbose):
 
         return dataset_organization, num_files, total_size, ds_creation_date
 
-    def _find_previous(self, relative_path, dataset_table, owner, owner_type):
+    def _find_previous(self, relative_path, owner, owner_type):
         """
-        Check to see if a dataset exists already in the registry, and if we are
-        allowed to overwrite it.
+        Find each dataset with combination of `relative_path`, `owner`,
+        `owner_type`.
+
+        We want to know, of those datasets, which are overwritable but have not
+        yet been marked as overwritten.
+
+        If any dataset with the same path has `is_overwritable=False`, the
+        routine returns None, indicating the dataset is not allowed to be
+        overwritten.
 
         Parameters
         ----------
         relative_path : str
             Relative path to dataset
-        dataset_table : SQLAlchemy Table object
-            Link to the dataset table
         owner : str
             Owner of the dataset
         owner_type : str
+            Owner type of the dataset
 
         Returns
         -------
-        previous : list
-            List of dataset IDs that are overwritable
+        dataset_id_list : list
+            List of dataset IDs that have the desired path combination that are
+            overwritable, but have not already previously been overwritten.
         """
 
         # Search for dataset in the registry.
-        stmt = (
-            select(dataset_table.c.dataset_id, dataset_table.c.is_overwritable)
-            .where(
-                dataset_table.c.relative_path == relative_path,
-                dataset_table.c.owner == owner,
-                dataset_table.c.owner_type == owner_type,
-            )
-            .order_by(dataset_table.c.dataset_id.desc())
+        dataset_table = self._get_table_metadata("dataset")
+        stmt = select(dataset_table)
```

Comment:
Shouldn't this be restricted to just the columns needed?

Reply:
Yes, have resolved